sdktests/client_side_events_summary.go

```diff
@@ -276,3 +282,147 @@ func doClientSideSummaryEventResetTest(t *ldtest.T) {
 		)),
 	)
 }
+
+func doClientSideSummaryBasicPrereqTest(t *ldtest.T) {
+	topLevelKey := "flag1"
+	topLevelResult := mockld.ClientSDKFlag{
+		Value:         ldvalue.String("value1-a"),
+		Variation:     o.Some(0),
+		FlagVersion:   o.Some(1),
+		Version:       11,
+		Prerequisites: []string{"prereq1", "prereq2"},
+	}
+
+	prereq1Key := "prereq1"
+	prereq1Result := mockld.ClientSDKFlag{
+		Value:         ldvalue.String("prereq1"),
+		Variation:     o.Some(0),
+		FlagVersion:   o.Some(1),
+		Version:       11,
+		Prerequisites: []string{"prereq3"},
+	}
+
+	prereq2Key := "prereq2"
+	prereq2Result := mockld.ClientSDKFlag{
+		Value:       ldvalue.String("prereq2"),
+		Variation:   o.Some(0),
+		FlagVersion: o.Some(1),
+		Version:     11,
+	}
+
+	prereq3Key := "prereq3"
+	prereq3Result := mockld.ClientSDKFlag{
+		Value:       ldvalue.String("prereq3"),
+		Variation:   o.Some(0),
+		FlagVersion: o.Some(1),
+		Version:     11,
+	}
+
+	contextA := ldcontext.New("user-a")
+	default1 := ldvalue.String("default1")
+	default2 := ldvalue.String("default2")
+	default3 := ldvalue.String("default3")
+
+	dataBuilder := mockld.NewClientSDKDataBuilder()
+	dataBuilder.Flag(topLevelKey, topLevelResult).
+		Flag(prereq1Key, prereq1Result).
+		Flag(prereq2Key, prereq2Result).
+		Flag(prereq3Key, prereq3Result)
+
+	dataSource := NewSDKDataSource(t, dataBuilder.Build())
+	events := NewSDKEventSinkWithGzip(t, t.Capabilities().Has(servicedef.CapabilityEventGzip))
+	client := NewSDKClient(t,
+		WithClientSideInitialContext(contextA),
+		dataSource, events)
+
+	_ = client.EvaluateFlag(t, servicedef.EvaluateFlagParams{FlagKey: prereq1Key, DefaultValue: default1})
```

Comment:
Okay @kinyoklion check out this setup I have. 

- I explicitly evaluate prereq1Key. This gets added to the summary event, including the default value I provided.
- I then evaluate top level, which has a prereq on prereqs 1 and 2. This results in prereq2 getting added to the summary event without a default value.
- We then evaluate prereq 2 explicitly, which just increments that value, but does not add the default value to the summary counter.